### PR TITLE
Feature/ti2023 13/login page(frontend)

### PR DESCRIPTION
**Pull request by bObXm**

### CODE REVIEW

File: EventPlannerAPI/EventPlannerAPI/appsettings.Local.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    "Host": "smtp.gmail.com",
+    "Port": 587
+  },
+  "FrontendBaseUrl": "localhost:3000",
+  "Jwt": {
+    "Key": "6AD2EFDE-AB2C-4841-A05E-7045C855BA22",
+    "Issuer": "https://localhost:7013",
+    "Audience": "https://localhost:7013"
+  }
+}
```
Comment (gheorghe-sava) [inline EventPlannerAPI/EventPlannerAPI/appsettings.Local.json]: This should not be committed to github, something is off with .gitignore

File: event-planner-frontend/src/features/login/store/thunks/logInThunk.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+import { LogInDto } from 'features/login/api/dtos';
+
+export const logInThunk = createAsyncThunk('logIn/logInThunk', async (userCredentials: LogInDto, thunkAPI) => {
+  try{
+    const request = await axios.post('https://localhost:7013/api/User/Login', userCredentials);
+    const response = await request.data;
```
Comment (gheorghe-sava) [inline event-planner-frontend/src/features/login/store/thunks/logInThunk.ts]: Why don't we use a client/service and use axios directly?

File: EventPlannerAPI/DataAccessLayer/Repositories/UserRepository.cs [hunk 3]
```diff
@@ -4,6 +4,7 @@
 using Microsoft.AspNetCore.Identity;
 using DataAccessLayer.Helpers;
 using Microsoft.AspNetCore.Http;
+using Microsoft.EntityFrameworkCore;
 
 namespace DataAccessLayer.Services
```
Comment (gheorghe-sava) [inline EventPlannerAPI/DataAccessLayer/Repositories/UserRepository.cs]: Out of context
We have to change the namespace to be consistent
```suggestion
namespace DataAccessLayer.Repositories
```

File: event-planner-frontend/src/features/login/store/slices/logInSlice.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { logInThunk } from '../thunks/logInThunk';
+
+const initialState={
```
Comment (gheorghe-sava) [inline event-planner-frontend/src/features/login/store/slices/logInSlice.ts]: Please define a type for the state

File: event-planner-frontend/src/features/login/store/slices/logInSlice.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    });
+    builder.addCase(logInThunk.fulfilled, (state, action) => {
+      state.loading = false;
+      state.user = action.payload;
+      state.error = null;
+    });
+    builder.addCase(logInThunk.rejected, (state, action) => {
+      state.loading = false;
+      state.user = null;
+      console.log(action.error.message);
```
Comment (gheorghe-sava) [inline event-planner-frontend/src/features/login/store/slices/logInSlice.ts]: We have to remove any console logs from production code

File: event-planner-frontend/src/features/login/store/selectors/logInSelectors.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+import { RootState } from 'redux/store';
+
+export const logInStateError = (state: RootState) => state.logIn.error;
+export const logInStateLoading = (state: RootState) => state.logIn.loading;
+export const logInStateUser = (state: RootState) => state.logIn.user;
```
Comment (gheorghe-sava) [inline event-planner-frontend/src/features/login/store/selectors/logInSelectors.ts]:
```suggestion
export const selectLogInStateError = (state: RootState) => state.logIn.error;
export const selectLogInStateLoading = (state: RootState) => state.logIn.loading;
export const selectLogInStateUser = (state: RootState) => state.logIn.user;
```

Comment (gheorghe-sava) [reply event-planner-frontend/src/features/login/store/selectors/logInSelectors.ts]: Please talk with the others to have this naming convention, any selector starts with select...

File: EventPlannerAPI/DataAccessLayer/Repositories/UserRepository.cs (hunk 3 above)
Comment (gheorghe-sava) [reply EventPlannerAPI/DataAccessLayer/Repositories/UserRepository.cs]: Are all DataAccessLayer.Services references updated?

Comment (bObXm) [reply EventPlannerAPI/DataAccessLayer/Repositories/UserRepository.cs]: yes

**Review by gheorghe-sava — APPROVED** [review]

File: EventPlannerAPI/DataAccessLayer/Repositories/UserRepository.cs (hunk 3 above)
Comment (bObXm) [reply EventPlannerAPI/DataAccessLayer/Repositories/UserRepository.cs]: [AdminPage](https://github.com/tecknoworks/SportEventManager/pull/13/commits/9ebed5b4b139788bdffc0bc2eb8ce6be59612def)

**Review by gheorghe-sava — APPROVED** [review]

**Review by HellHound97 — APPROVED** [review]

**Pull request merged**